function_calling/notebooks/function_calling_guide.ipynb

```diff
@@ -0,0 +1,1216 @@
+      " '{\\n'\n",
+      " '  \"name\": \"python_repl\",\\n'\n",
+      " '  \"description\": \"A Python shell. Use this to execute python commands. Input '\n",
+      " 'should be a valid python commands and expressions. If you want to see the '\n",
+      " 'output of a value, you should print it out with `print(...), if you need a '\n",
+      " 'specific module you should import it`.\",\\n'\n",
+      " '  \"properties\": {\\n'\n",
+      " '    \"command\": {\\n'\n",
+      " '      \"title\": \"Command\",\\n'\n",
+      " '      \"description\": \"python code to execute to evaluate\",\\n'\n",
+      " '      \"type\": \"string\"\\n'\n",
+      " '    }\\n'\n",
+      " '  }\\n'\n",
+      " '}\\n'\n",
+      " '{\\n'\n",
+      " '  \"name\": \"ConversationalResponse\",\\n'\n",
+      " '  \"description\": \"Respond conversationally only if no other tools should be '\n",
+      " 'called for a given query, or if you have a final answer.\",\\n'\n",
+      " '  \"properties\": {\\n'\n",
+      " '    \"response\": {\\n'\n",
+      " '      \"title\": \"Response\",\\n'\n",
+      " '      \"description\": \"Conversational response to the user.\",\\n'\n",
+      " '      \"type\": \"string\"\\n'\n",
+      " '    }\\n'\n",
+      " '  }\\n'\n",
+      " '}')\n"
+     ]
+    }
+   ],
+   "source": [
+    "tools_schemas = get_tools_schemas(tools, default = ConversationalResponse)\n",
+    "tools_schemas = \"\\n\".join([json.dumps(tool, indent=2) for tool  in tools_schemas])\n",
+    "pprint(tools_schemas)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Function Calling "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### LLM definition"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 61,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "llm = Sambaverse(\n",
+    "    sambaverse_model_name=\"Meta/Meta-Llama-3-70B-Instruct\",\n",
+    "    model_kwargs={       \n",
+    "            \"max_tokens_to_generate\": 2048,\n",
+    "            \"select_expert\": \"Meta-Llama-3-70B-Instruct\",\n",
+    "            \"process_prompt\": True,\n",
+    "            \"temperature\": 0.01\n",
+    "        }\n",
+    ")\n",
+    "\n",
+    "# llm= SambaStudio(\n",
+    "#     model_kwargs={       \n",
+    "#             \"max_tokens_to_generate\": 2048,\n",
+    "#             \"select_expert\": \"Meta-Llama-3-8B-Instruct\",\n",
+    "#             \"process_prompt\": False\n",
+    "#         }\n",
+    "# )\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Tool execution"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 62,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tools_map = {\n",
+    "    \"get_time\": get_time, \n",
+    "    \"calculator\": calculator, \n",
+    "    \"python_repl\": python_repl,\n",
+    "}\n",
+    "\n",
+    "def execute(tools):\n",
```

Comment:
Maybe a brief description of what this functions does?

Reply:
added